modules/caddyhttp/responsewriter.go

```diff
@@ -240,6 +242,12 @@ func (rr *responseRecorder) FlushError() error {
 	return nil
 }
 
+// Private interface so it can only be used in this package
+// #TODO: maybe export it later
+func (rr *responseRecorder) setReadSize(size *int) {
```

Comment:
Passing in a `*int` is a bit weird. Can we just pass in a `int`? Any reason we need a pointer at all, actually?

Reply:
This pointer points to the member of the wrapped request body. So when it's updated, the corresponding log entry is updated as well. `int` requires more changes to the code.

Reply:
Ah, I see. Is that thread-safe? (Would it ever be used across goroutines?)

Reply:
The `net.Conn` interface is not thread-safe by default unless specified explicitly. But the only race I know of is when caddy [sends websocket goaway frame](https://github.com/caddyserver/caddy/blob/master/modules/caddyhttp/reverseproxy/streaming.go#L114), which is not thread-safe in the first place. But clients are expected to reconnect anyway, the stats are a bit off (goaway frames are small), nothing breaking.